test/scripts/common.inc

```diff
@@ -45,3 +47,30 @@ identity_user() {
 identity_system() {
     "${XRHIDGEN}" -org-id "${ORG_ID}" system -cn "6f324116-b3d2-11ed-8a37-482ae3863d30" -cert-type system | base64nowrap
 }
+
+identity_service_account() {
+    "${XRHIDGEN}" -org-id "${ORG_ID}" service-account -client-id "${XRHID_CLIENT_ID}" -username test | base64nowrap
```

Comment:
The command should be:
```
"${XRHIDGEN}" -org-id "${ORG_ID}" -auth-type jwt-auth service-account -client-id "${XRHID_CLIENT_ID}" -username test | base64nowrap
```
I.e the `-auth-type jwt-auth` was added.

The reason is that when I tried e.g. `XRHID_AS=service-account ./test/scripts/local-domains-list.sh` I got 401 issue.  

Logs did not help me much as there was only error from user predicate and not the service account predicate.

So after small modification:
```go
func NewEnforceOr(predicates ...IdentityPredicate) IdentityPredicate {
	return func(data *identity.XRHID) error {
		var allErrs error
		for i := range predicates {
			if err := predicates[i](data); err == nil {
				return nil
			} else {
				allErrs = errors.Join(allErrs, err)
			}
		}
		return allErrs
	}
}
```

I now see issue:
```
level=ERROR msg="'user-and-sa-identity' IdentityPredicate failed: 'Identity.Type=ServiceAccount' is not 'User'\n'Identity.AuthType' is not 'jwt-auth'"
```

A Q can be also whether we must limit it to `jwt-auth`.

Reply:
thanks for this; I like the idea to join the errors and provide full error information. Adding that. Considering the `jwt-auth` I have realized I missed some changes in the scripts (adding the suggested to specify  `AUTH_TYPE`).

IMHO we would need to limit to `jwt-auth` (user and service-account) and `cert-auth`; Adding the change too.

Reply:
Might be good to do the additional changes in separate PR. Here we can just fix the `identity_service_account` and merge.

Reply:
I have added additional changes at: https://github.com/podengo-project/idmsvc-backend/pull/249 but I had some situation with different golang versions (1.21 and 1.22) when running the unit tests, that evoked different results for the echo router unit-tests.